sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -38,6 +39,11 @@ import org.apache.spark.unsafe.types.UTF8String.{IntWrapper, LongWrapper}
 
 object Cast {
 
+  /**
+   * A tag to decide if a CAST is generated during analysis.
+   */
+  val AUTO_GENERATED_TAG = new TreeNodeTag[Boolean]("auto-generated-cast")
```

Comment:
it's a bit weird to have a property that can only be set to false. How about `user_specified_cast` that can only be set to  true?

Reply:
Looks better.